### PR TITLE
runtime: add linux default fs validation

### PR DESCRIPTION
**Pull request by Mashimiao**
Signed-off-by: Ma Shimiao <mashimiao.fnst@cn.fujitsu.com>

### CODE REVIEW

File: cmd/runtimetest/main.go [hunk 1]
```diff
@@ -229,6 +238,28 @@ func validateRootFS(spec *rspec.Spec) error {
...
+	}
+
+	mountsMap := make(map[string]string)
+	for _, mountInfo := range mountInfos {
+		mountsMap[mountInfo.Mountpoint] = mountInfo.Fstype
+	}
+
+	for fs, fstype := range defaultFS {
+		if !(mountsMap[fs] == fstype) {
+			return fmt.Errorf("%v must exists and expected type is %v", fs, fstype)
```
Comment (wking) [inline cmd/runtimetest/main.go]: nit: “exists” → “exist”.

**Comment by wking** [comment]
On Thu, Jun 23, 2016 at 09:11:21AM -0700, Ma Shimiao wrote:
>   * runtime: add linux default fs validation

This looks ok to me, although there are fuzzy corner cases if
config.json specifies a different type be mounted over one of these
locations.  See the last point in [1] about when the runtime is
supposed to supply these mounts/devices, and we still have no wording
about that in the spec itself.

[1]: https://github.com/opencontainers/runtime-spec/pull/164#issuecomment-138984824

File: cmd/runtimetest/main.go (hunk 1 above)
Comment (Mashimiao) [inline cmd/runtimetest/main.go]: fixed, thanks.

**Comment by Mashimiao** [comment]
As smoe validations related to linux os, splited linux-specific valdiations

**Comment by wking** [comment]
On Fri, Jun 24, 2016 at 07:16:22PM -0700, Ma Shimiao wrote:
> As smoe validations related to linux os, splited linux-specific
> valdiations

258adc81 itself looks good for me (except for “valdations” →
“validations” in the commit message).

I'm still not convinced that the current spec wording is enforcable
[1], but I'm ok with punting on that for now (until someone asks us to
test the case where a config.json asks for a bind mount over /tmp? ;).

[1]: https://github.com/opencontainers/ocitools/pull/121#issuecomment-228115790

**Comment by Mashimiao** [comment]
fixed valdations->validations

**Comment by wking** [comment]
On Sat, Jun 25, 2016 at 01:12:52AM -0700, Ma Shimiao wrote:
> fixed valdations->validations

798d84f0 looks good to me if we're punting on the corner cases [1].

[1]: https://github.com/opencontainers/ocitools/pull/121#issuecomment-228115790

File: cmd/runtimetest/main.go [hunk 2]
```diff
@@ -22,6 +22,15 @@ import (
 // the kernel
 const PR_GET_NO_NEW_PRIVS = 39
 
+var (
+	defaultFS = map[string]string{
+		"/proc":   "proc",
+		"/sys":    "sysfs",
+		"dev/pts": "devpts",
```
Comment (liangchenye) [inline cmd/runtimetest/main.go]: Change "dev/pts" to /dev/pts ?

same with "dev/shm"

Comment (Mashimiao) [inline cmd/runtimetest/main.go]: fixed

**Comment by Mashimiao** [comment]
ping @liangchenye  @mrunalp

**Comment by liangchenye** [comment]
e4b2183  LGTM

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**